### PR TITLE
sheesy-cli: init at 4.0.11

### PR DESCRIPTION
**Pull request by devhell**
I don't have the means to test this on Darwin, but I've added it provisionally.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121531` at f0105e98 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>sheesy-cli</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/tools/security/sheesy-cli/default.nix:45:5:
```
   |
45 |     license = with licenses; [ lgpl21 ];
   |     ^
```
</li>
  </ul>
</details>

**Comment by devhell** [comment]
@grahamcofborg build

**Comment by devhell** [comment]
@ofborg build

File: pkgs/tools/security/sheesy-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+, libgpgerror, gpgme, gettext, openssl, darwin
+}:
+
+let
+
+  features = [ "vault" "extract" "completions" "substitute" "process" ];
+
+in
+
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/security/sheesy-cli/default.nix:11]:
```suggestion
rustPlatform.buildRustPackage rec {
```

File: pkgs/tools/security/sheesy-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  src = fetchCrate {
+    inherit version pname;
+    sha256 = "1l21ji9zqy8x1g2gvqwdhya505max07ibx1hh88s36k0jbvdb7xc";
+  };
+
+  cargoSha256 = "159a5ph1gxwcgahyr8885lq3c1w76nxzfrfdpyqixqrr7jzx2rd3";
+  cargoDepsName = pname;
+
+  cargoBuildFlags = [ "--bin sy" "--features" "${lib.concatStringsSep "," features}" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/security/sheesy-cli/default.nix:23]:
```suggestion
  cargoBuildFlags = [ "--bin sy" "--features" "vault,extract,completions,substitute,process" ];
```

File: pkgs/tools/security/sheesy-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  nativeBuildInputs = [ libgpgerror gpgme gettext installShellFiles ];
+
+  buildInputs = [ ]
+  ++ (lib.optionals stdenv.isLinux [ openssl ])
+  ++ (lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+    Security
+  ]));
```
Comment (SuperSandro2000) [inline pkgs/tools/security/sheesy-cli/default.nix:31]:
```suggestion
  buildInputs = lib.optionals stdenv.isLinux [ openssl ]
    ++ lib.optionals stdenv.isDarwin [ Security ];
```
Please inherit Security from the top level entry.

File: pkgs/tools/security/sheesy-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  postInstall = ''
+    for shell in bash fish zsh; do
+      $out/bin/sy completions $shell > sy.$shell
+      installShellCompletion sy.$shell
+    done
```
Comment (SuperSandro2000) [inline pkgs/tools/security/sheesy-cli/default.nix:37]:
```suggestion
    installShellCompletion --cmd sy \
      --bash <($out/bin/sy completions bash)
      --fish <($out/bin/sy completions fish)
      --zsh <($out/bin/sy completions zsh)
```

File: pkgs/tools/security/sheesy-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  '';
+
+  meta = with lib; {
+    description = "The 'share-secrets-safely' CLI to interact with GPG/pass-like vaults";
+    homepage = "https://share-secrets-safely.github.io/cli/";
+    changelog = "https://github.com/share-secrets-safely/cli/releases/tag/${version}";
+    maintainers = with maintainers; [ devhell ];
+    license = with licenses; [ lgpl21Only ];
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/tools/security/sheesy-cli/default.nix:46]:
```suggestion
```

buildRustPackage defaults to this

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request merged**